core/src/avm2/object.rs

```diff
@@ -611,27 +576,43 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
     /// Call a method by its index.
     ///
     /// This directly corresponds with the AVM2 operation `callmethod`.
-    #[allow(unused_mut)] //Not unused.
     fn call_method(
-        mut self,
+        self,
         id: u32,
         arguments: &[Value<'gc>],
         activation: &mut Activation<'_, 'gc>,
     ) -> Result<Value<'gc>, Error<'gc>> {
-        if self.get_bound_method(id).is_none() {
-            if let Some(vtable) = self.vtable() {
-                if let Some(bound_method) = vtable.make_bound_method(activation, self.into(), id) {
-                    self.install_bound_method(activation.context.gc_context, id, bound_method);
-                }
-            }
+        if let Some(bound_method) = self.get_bound_method(id) {
+            return bound_method.call(Value::Undefined, arguments, activation);
```

Comment:
I'm not sure if passing `Undefined` as the receiver is valid here, but I believe it should be as `unbound_receiver` seems unused for bound methods, but it would be nice if someone could confirm!

Reply:
Can we pass the actual receiver just for consistency?

Reply:
Yes, this was a blind optimization anyway and on review, seems to impact codegen minimally